### ISSUE TITLE
GENDARME package can not be upgraded - missing dependencies

### ISSUE THREAD

**Issue by cement-head** [mono/mono#18788]
## Steps to Reproduce

1. Install 18.04.3 LTS
2. Add MONO-PROJECT Repos
3. Run Update & Upgrade

### Current Behavior

Large amount of packages upgraded, except for one:

```
$ sudo apt list --upgradable
gendarme/stable-bionic,stable-bionic 4.2-3xamarin3+ubuntu1804b1 all [upgradable from: 4.2-2.2]
N: There is 1 additional version. Please use the '-a' switch to see it

$ sudo apt list --upgradable -a
Listing... Done
gendarme/stable-bionic,stable-bionic 4.2-3xamarin3+ubuntu1804b1 all [upgradable from: 4.2-2.2]
gendarme/bionic,bionic,now 4.2-2.2 all [installed,upgradable to: 4.2-3xamarin3+ubuntu1804b1]
```

### Expected Behavior

Upgrade of all packages; no holds due to dependencies

Missing dependency(ies) are not listed.  Package is simply held - no errors.

## On which platforms did you notice this

[ ] macOS
[X ] Linux
[ ] Windows

**Version Used**:

$ mono --version
Mono JIT compiler version 6.8.0.105 (tarball Tue Feb  4 21:20:20 UTC 2020)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           __thread
	SIGSEGV:       altstack
	Notifications: epoll
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug 
	Interpreter:   yes
	LLVM:          yes(610)
	Suspend:       hybrid
	GC:            sgen (concurrent by default)

## Stacktrace

N/A

**Comment by directhex** [comment]
Hm. There's an ages-old hack in place to skip that dependency, but it's not working? Will check the build logs.

**Comment by directhex** [comment]
Fixed in `4.2-3xamarin4`
